### PR TITLE
Added first cut of shared secret and shared secret type

### PR DESCRIPTION
**Pull request by dave-cheong**
Adds these new entities to the model and hooked up to identity and profile. These have been coded from a modelling perspective only. No real business logic have been applied to these yet. If required, they will come later as a separate pull request.

Specifically, no implementation or consideration of:
- what's conditionally mandatory
- what's valid based on the specialised Identity types
- how the Identity, Profile, Name, Shared Secret will be used

### CODE REVIEW

File: backend/typescript/models/sharedSecretType.model.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        })
+        .exec();
+});
+
+SharedSecretTypeSchema.static('findByCodeInDateRange', (code:String) => {
+    return this.SharedSecretTypeModel
+        .findOne({
+            code: code,
+            startDate: {$lte: new Date()},
+            $or: [{endDate: null}, {endDate: {$gt: new Date()}}]
```
Comment (sundriver) [inline backend/typescript/models/sharedSecretType.model.ts]: shouldn't it be $gte ?
Also end date must be end of today not now

Comment (dave-cheong) [inline backend/typescript/models/sharedSecretType.model.ts]: I've added a technical item to the backlog for your comment above. I'll address them in a separate pull request, as it affects all the ICodeDecode entities. So keep this pull request for the new shared secret and type entities only. If you're cool with that, please accept and I'll work on the date comments above separately.

Comment (dave-cheong) [inline backend/typescript/models/sharedSecretType.model.ts]: Definitely for the gte case, but do you know i the endDate is time constraint too? So can we expire something in the middle of the day? @sundriver

File: backend/typescript/models/identity.model.ts [hunk 2]
```diff
@@ -61,15 +156,36 @@ const IdentitySchema = RAMSchema({
...
-    token: {
+    agencyToken: {
         type: String,
         trim: true
     },
-    scheme: {
+    invitationCodeStatus: {
+        type: String,
+        trim: true,
+        enum: IdentityInvitationCodeStatus.valueStrings()
```
Comment (sundriver) [inline backend/typescript/models/identity.model.ts]: required:true

File: backend/typescript/models/identity.model.ts [hunk 3]
```diff
@@ -61,15 +156,36 @@ const IdentitySchema = RAMSchema({
...
+    },
+    invitationCodeExpiryTimestamp: {
```
Comment (sundriver) [inline backend/typescript/models/identity.model.ts]: required: true

File: backend/typescript/models/sharedSecretType.model.ts (hunk 1 above)
Comment (sundriver) [inline backend/typescript/models/sharedSecretType.model.ts]: Date shouldn't be hard coded in any date valitiy check, it should come in as an argument

Comment (dave-cheong) [inline backend/typescript/models/sharedSecretType.model.ts]: The find/list in date range functions now all accept date as input.

Comment (dave-cheong) [inline backend/typescript/models/sharedSecretType.model.ts]: Changed to gte

Comment (dave-cheong) [inline backend/typescript/models/sharedSecretType.model.ts]: Introduced RAMEnum in base.ts

**Pull request merged**